src/inspect_ai/model/_providers/_openai_batch.py

```diff
@@ -212,11 +229,54 @@ async def _handle_batch_result_file(
                 )
 
             # Store the result in the dictionary instead of sending to result_stream
-            results[request_id] = (
-                ChatCompletion.model_validate(result["response"]["body"])
-                if (error := result.get("error")) is None
-                else self._openai_client._make_status_error_from_response(  # pyright: ignore[reportPrivateUsage]
-                    httpx.Response(status_code=error["code"], text=error["message"])
+            if (error := result.get("error")) is None:
+                response_body = result["response"]["body"]
+                results[request_id] = self._response_cls.model_validate(response_body)
+            else:
+                results[request_id] = (
+                    self._openai_client._make_status_error_from_response(  # pyright: ignore[reportPrivateUsage]
+                        httpx.Response(status_code=error["code"], text=error["message"])
+                    )
                 )
-            )
         return results
+
+    def _results_from_rejection(
```

Comment:
This is the code that extracts that 2nd type of error reporting and packages it in a way that can be passed to the base class for reject all of the promises in the batch using shared/existing code.